src/main/ruby/core/main.rb

```diff
@@ -70,13 +70,15 @@ def to_s
 }
 
 Truffle::Boot.delay do
-  Signal.trap('INT') do
-    if Truffle::Boot.get_option('backtraces.on_interrupt')
-      puts 'Interrupting...'
-      show_backtraces.call
-    end
+  if Truffle::Boot.get_option('platform.handle_interrupt')
+    Signal.trap('INT') do
+      if Truffle::Boot.get_option('backtraces.on_interrupt')
+        puts 'Interrupting...'
+        show_backtraces.call
+      end
 
-    raise Interrupt
+      raise Interrupt
+    end
```

Comment:
FWIW there is also a default handler for SIGALRM below. But if it doesn't cause problems it's probably fine to leave it.